src/AutorestSwift/View Models/RequestViewModel.swift

```diff
@@ -26,35 +26,54 @@
 
 import Foundation
 
+enum RequestBodyType: String {
+    /// Service requires some kind of serialized object
+    case body
+    /// Service requires no request body
+    case noBody
+    /// Service requires a serialzed JSON merge-patch body
+    case patchBody
+}
+
+struct BodyParam {
+    let name: String
+    let type: String
+    let propertyNames: [String]
+
+    init(from parameter: Parameter) {
+        assert(belongsInSignature(param: parameter), "Body param found that doesn't go in signature...")
+        self.name = parameter.name
+        self.type = parameter.schema.name
+        self.propertyNames = (parameter.schema.properties ?? []).map { $0.name }
+    }
+}
+
 /// View Model for the method request creation.
 struct RequestViewModel {
     let path: String
     let method: String
-    let objectType: String?
-    let objectName: String?
-    let hasBody: Bool
+    let bodyParam: BodyParam?
+    /// Identifies the correct snippet to use when rendering the view model
+    let strategy: String
 
     init(from request: Request) {
         // load HttpRequest properties
         let httpRequest = request.protocol.http as? HttpRequest
         self.path = httpRequest?.path ?? ""
         self.method = httpRequest?.method.rawValue ?? ""
 
-        // load HttpWithBodyRequest specfic properties
-        let httpWithBodyRequest = request.protocol.http as? HttpWithBodyRequest
-
-        // TODO: only support the first signature parameter in Reqest now
+        // TODO: only support the first signature parameter in Request now
         let bodyParams = getBodyParameters(signatureParameters: request.signatureParameters)
         assert(bodyParams.count <= 1, "Unexpectedly found more than 1 body parameters in request... \(request.name)")
-
         let bodyParam = bodyParams.first
-        self.objectType = bodyParam?.schema.name
-        if belongsInSignature(param: bodyParam) {
```

Comment:
So if we don't check whether the 'body' is from signature or from options, the function  belongsInSignature() can be removed

Reply:
I thought the logic for options was anything that was a query or header param, which means we wouldn't expect to ever find the body param in the options?

Reply:
Yes. you are right.  belongsInSignature can be removed.